src/common/api/hive.ts

```diff
@@ -599,6 +601,24 @@ export interface BlogEntry {
 export const getBlogEntries = (username: string, limit: number = dataLimit): Promise<BlogEntry[]> =>
   client.call("condenser_api", "get_blog_entries", [username, 0, limit]);
 
+export const getAccountVotesTrail = (
+  username: string,
+  start = -1,
+  limit = 20
+): Promise<BlogEntry[]> => {
+  let params = [username, start, limit, 1];
+  return client.call("condenser_api", "get_account_history", params).then((data) => {
+    let result = data
+      .map((obj: any) => {
+        return { ...obj[1].op[1], num: obj[0] };
+      })
+      .filter((obj: any) => obj.voter === username);
+    return Promise.all(
+      result.map((obj: any) => getPostNew(obj.author, obj.permlink, username, obj.num))
+    );
+  });
```

Comment:
1. Need to rewrite as `async/await`;
2. `Promise.all` uses to resolve multiple promises together but in this case you have one promise with array object. Make as in first item and just return your data

Reply:
@dkildar  I didn't understand that. 

That function `getAccountVotesTrail` first fetches from the account history what are the posts voted, but still the posts are not resolved, they are in the format 

```
{
  "voter": "igormuba-dev",
  "author": "hispapro",
  "permlink": "soporte-diario-939-publicaciones-valoradas",
  "weight": 10000,
  "num": 142
}
```

So the function maps each of those  to the promise `getPostNew` so each entry in that array is a post promise that will be fetched separately, that is why the `Promise.all` because it is an array of post resolutions, so the return from that function is the posts already resolved

If that doesn't look like a good solution could you elaborate on your suggestion? I couldn't exactly understand what could be improved there. Thanks.

Reply:
You're right. My fault:)